File: src/main/java/it/pagopa/ecommerce/config/JwtHeaderFactory.java
package it.pagopa.ecommerce.config;

import it.pagopa.ecommerce.services.TokenService;
import jakarta.inject.Inject;
import jakarta.ws.rs.core.MultivaluedMap;
import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory;

import java.util.UUID;

public class JwtHeaderFactory implements ClientHeadersFactory {

    @Inject
    TokenService tokenService;

    @Override
    public MultivaluedMap<String, String> update(
            MultivaluedMap<String, String> incomingHeaders,
            MultivaluedMap<String, String> clientOutgoingHeaders
    ) {
        //String token = tokenService.getToken();
        //clientOutgoingHeaders.add("Authorization", "Bearer " + token);
        clientOutgoingHeaders.add("X-request-uuid", UUID.randomUUID().toString());
        return clientOutgoingHeaders;
    }
}

File: src/main/kotlin/ecommerce/config/JwtHeaderFactory.kt
package ecommerce.config

import ecommerce.services.TokenService
import jakarta.inject.Inject
import jakarta.ws.rs.core.MultivaluedMap
import java.util.UUID
import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory

class JwtHeaderFactory : ClientHeadersFactory {

    @Inject lateinit var tokenService: TokenService

    override fun update(
        incomingHeaders: MultivaluedMap<String, String>?,
        clientOutgoingHeaders: MultivaluedMap<String, String>?,
    ): MultivaluedMap<String, String>? {
        val token = tokenService.getToken()
        clientOutgoingHeaders?.add("Authorization", "Bearer $token")
        clientOutgoingHeaders?.add("X-request-uuid", UUID.randomUUID().toString())
        return clientOutgoingHeaders
    }
}
